lib/Doctrine/ORM/ORMInvalidArgumentException.php

```diff
@@ -198,4 +198,20 @@ private static function objToStr($obj)
     {
         return method_exists($obj, '__toString') ? (string)$obj : get_class($obj).'@'.spl_object_hash($obj);
     }
+
+    /**
+     * @return ORMInvalidArgumentException
+     */
+    public static function invalidAssociation($relation, $fieldname, $value) {
+        return new self('Expected an Object for relation '.get_class($relation).'::'.$fieldname.' got '.gettype($value).' instead.');
+    }
+
+    /**
+     * @return ORMInvalidArgumentException
+     */
+    public static function invalidAssociation($entry) {
+        $ex = new self(gettype($entry) . ' is not an Object.');
+        $ex->value = $entry;
```

Comment:
Don't set a dynamic property: simply limit yourself to inlining it *IF* it is a scalar type